### PR TITLE
USEID-161: Propose via adr a csp for embedding the iframe

### PR DESCRIPTION
**Pull request by code28**

### CODE REVIEW

File: doc/adr/0005-csp-for-embedding-widget.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+## Decision
+
+We use the [`frame-ancestors` directive at the CSP header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/frame-ancestors). The browsers only allow pages to embed an iframe, whose domain is listed there.
+
+We set this header only to the one domain, which is actually embedding the iframe and don't provide a full list of all allowed domains.
+
+The page inside the iframe can't get the domain of the parent page. Therefore the [widget.js file](0004-embedding-widget-via-js-wrapper.md) adds the domain of the parent page as query parameter to the iframe url.
+
+The backend can check the domain from the query parameter with the list of registered domains and if the domain is found, it is added into the CSP header.
```
Comment (pgurusinga) [inline doc/adr/0005-csp-for-embedding-widget.md:25]: should we decide where do we store the lists of allowed domain?

File: doc/adr/0005-csp-for-embedding-widget.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+We evaluated using a kind of api key (which would then be public), but creating a mapping of domain to api key would be quite easy, so we use the domain name as quasi public api key.
+
+## Consequences
+
+Only the registered domains are allowed to embed widgets and this behaviour is enforced by the browsers, when a CSP header with `frame-ancestors` is present.
+
+By adding the parent domain as query parameter, the server can check this domain in the list. We therefore need to have a list of registered domains in the backend server.
+
+It's not possible to MITM this embedding (in default browsers). The domain can't be chosen arbitrarily, since we check it in the list of registered domains on the backend server.
```
Comment (pgurusinga) [inline doc/adr/0005-csp-for-embedding-widget.md:35]: as far as I understand, by adding `frame-src` and `frame-ancestor` we create 1:1 direct connection between client and server and therefor MITM won't be possible?

**Review by pgurusinga — APPROVED** [review]
just some question for CSP, otherwise LGTM

File: doc/adr/0005-csp-for-embedding-widget.md (hunk 1 above)
Comment (code28) [reply doc/adr/0005-csp-for-embedding-widget.md:25]: I'd say, that is irrelevant and out of the scope for that adr.

File: doc/adr/0005-csp-for-embedding-widget.md (hunk 2 above)
Comment (code28) [reply doc/adr/0005-csp-for-embedding-widget.md:35]: That's what I wanted to say with that, but I can add another sentence to make it even clearer 👍

**Pull request closed without merging**